### DISCUSSION TITLE
Limited data in Grafana

### DISCUSSION THREAD

**Discussion by DorDana91** [teslamate-org/teslamate#3550] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

After I installed the apps, I could see in both Grafana and DB that I had limited data.

### Relevant log output

```shell
log
```

### Type of installation

Docker

### Version

latest

**Comment by cwanja** [comment]
Going to need a lot more information, but let's start with what does "limited data" mean? Screenshots and logs help here. 

If you _just_ installed the apps, there will be near zero data there until you connect to your Tesla account and start scraping data.

**Comment by DorDana91** [comment]
I ran it locally, and visited localhost:4000 and now I need to fill the tokens.
I've generated tokens and succeed to log in, but I can see a very small amount of data

**Comment by cwanja** [comment]
To you, what is "a very small amount of data". 

From the steps you have provided, sounds expected. Go take a drive or charge the vehicle and see if the data is collected. 

Moving to discussion for now.

**Comment by DorDana91** [comment]
see my logs:

mytesla-teslamate-1  | 2023-12-06 18:54:15.196 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/XXXXXXXXXXX/vehicle_data -> 408 (9135.951 ms)
mytesla-teslamate-1  | 2023-12-06 18:54:15.196 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_description" => "", "response" => nil}

**Reply by cwanja** [reply]
Timeouts are normal and do not indicate an error. 

What data do you see on TeslaMate's homepage? Additionally, please perform a test drive or a charge and see if data is collected.

**Comment by DorDana91** [comment]
charge and drives tables are empty

**Reply by cwanja** [reply]
Until you drive or charge, that is expected. Please see previous posts.

**Comment by DorDana91** [comment]
but how i can see my battery stats?

**Reply by cwanja** [reply]
https://docs.teslamate.org/docs/faq#why-are-no-consumption-values-displayed-in-grafana

You need to perform drives and charges to get consumption and battery detail. 

What "battery stats" were you expecting to just show up by installing TeslaMate?

**Reply by DorDana91** [reply]
it can't fetch all the data that I already have before installing the app?

**Reply by cwanja** [reply, marked as answer]
No. The API does not offer historical data. No third-party app can do that. 

TeslaMate only pulls the data from the moment you connect your Tesla account. And if your self-installed TeslaMate service is down, it will not backfill the data when you start it back up. Tesla's API does not provide historical data or the ability to fill gaps in data.

**Reply by DorDana91** [reply]
Thanks a lot
